### PR TITLE
[connection graph] add baud rate config to console_links.csv

### PR DESCRIPTION
**Pull request by Xichen96**
### Description of PR

Add baud rate config to console_links.csv and the related tools.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?
Automating deployment of MC0 would need this information.

#### How did you do it?
Add a column to console_links.csv. Also conn_graph_facts.py need adjustments to include this information.

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>ansible/library/conn_graph_facts.py:28:14: E225 missing whitespace around operator<br/>ansible/library/conn_graph_facts.py:73:9: E225 missing whitespace around operator<br/>ansible/library/conn_graph_facts.py:116:121: E501 line too long (121 > 120 characters)<br/>ansible/library/conn_graph_facts.py:190:121: E501 line too long (126 > 120 characters)<br/>ansible/library/conn_graph_facts.py:207:84: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:207:121: E501 line too long (180 > 120 characters)<br/>ansible/library/conn_graph_facts.py:209:121: E501 line too long (181 > 120 characters)<br/>ansible/library/conn_graph_facts.py:244:68: E203 whitespace before ':'<br/>ansible/library/conn_graph_facts.py:248:40: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:249:39: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:254:66: E203 whitespace before ':'<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: ansible/files/sonic_lab_console_links.csv [hunk 1]
```diff
@@ -1,4 +1,4 @@
-StartDevice,StartPort,EndDevice,Console_type,Proxy
-console-1,10,str-msn2700-01,ssh,root
-console-1,11,str-7260-10,ssh,root
-console-1,12,str-7260-11,ssh,root
\ No newline at end of file
+StartDevice,StartPort,EndDevice,Console_type,Proxy,BaudRate
+console-1,10,str-msn2700-01,ssh,root,
```
Comment (Blueve) [inline ansible/files/sonic_lab_console_links.csv:2]: Let's add sample baudrate at the end, e.g. 
```
console-1,10,str-msn2700-01,ssh,root,9600
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>ansible/library/conn_graph_facts.py:28:14: E225 missing whitespace around operator<br/>ansible/library/conn_graph_facts.py:73:9: E225 missing whitespace around operator<br/>ansible/library/conn_graph_facts.py:116:121: E501 line too long (121 > 120 characters)<br/>ansible/library/conn_graph_facts.py:190:121: E501 line too long (126 > 120 characters)<br/>ansible/library/conn_graph_facts.py:207:84: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:207:121: E501 line too long (180 > 120 characters)<br/>ansible/library/conn_graph_facts.py:209:121: E501 line too long (181 > 120 characters)<br/>ansible/library/conn_graph_facts.py:244:68: E203 whitespace before ':'<br/>ansible/library/conn_graph_facts.py:248:40: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:249:39: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:254:66: E203 whitespace before ':'<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by Xichen96** [comment]
Test deployment with console_links.csv with no BaudRate column, with BaudRate column but no value, and with BaudRate column and with value, all turn out fine.

**Review by Blueve — APPROVED** [review]

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>ansible/library/conn_graph_facts.py:28:14: E225 missing whitespace around operator<br/>ansible/library/conn_graph_facts.py:73:9: E225 missing whitespace around operator<br/>ansible/library/conn_graph_facts.py:116:121: E501 line too long (121 > 120 characters)<br/>ansible/library/conn_graph_facts.py:190:121: E501 line too long (126 > 120 characters)<br/>ansible/library/conn_graph_facts.py:207:84: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:207:121: E501 line too long (180 > 120 characters)<br/>ansible/library/conn_graph_facts.py:209:121: E501 line too long (181 > 120 characters)<br/>ansible/library/conn_graph_facts.py:244:68: E203 whitespace before ':'<br/>ansible/library/conn_graph_facts.py:248:40: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:249:39: E231 missing whitespace after ':'<br/>ansible/library/conn_graph_facts.py:254:66: E203 whitespace before ':'<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by lizhijianrd** [comment]
Hi @Xichen96, since you didn't update `creategraph.py` in this PR, how can we generate a conn-graph.xml with band info? Are you plan to support that in a separate PR?

**Comment by Xichen96** [comment]
> Hi @Xichen96, since you didn't update `creategraph.py` in this PR, how can we generate a conn-graph.xml with band info? Are you plan to support that in a separate PR?

We don't need extra support, creategraph reads the added columns itself. The change in lab_connection_graph.xml in this pr is updated by the original creategraph script.

**Pull request merged**